### ISSUE TITLE
Wrong Python indentation when a for loop expression contains a lambda

### ISSUE THREAD

**Issue by aldanor** [jupyterlab/jupyterlab#5842]
If the iterator expression contains a lambda (which contains a colon, which is probably the problem), the consequent indentation is all wrong; e.g.:

```python
for k, v in sorted(d.items(), key=lambda v: -len(v)):
pass
```

**Comment by jasongrout** [comment]
Indeed, it looks like raw codemirror has the same issue: https://codemirror.net/mode/python/index.html

Do you mind opening an issue in the codemirror repo? https://github.com/codemirror/CodeMirror
